Linux/install-vrcx.sh

```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 
 steamapps=$HOME/.local/share/Steam/steamapps/compatdata
-stable=$(curl -s https://api.github.com/repos/vrcx-team/VRCX/releases/latest | grep -o "https.*releases/download.*zip")
-nightly=$(curl -s https://api.github.com/repos/Natsumi-sama/VRCX/releases/latest | grep -o "https.*releases/download.*zip")
+stable=$(curl -s https://api0.vrcx.app/releases/stable/latest | grep -o "https.*releases/download.*zip")
```

Comment:
What's this change for? `api0.vrcx.app` seemingly just proxies the release endpoint. The bash script has no cryptographic / hash verification for the binaries themselves, I'm kinda queasy about moving this to a server without published source code.

Reply:
https://github.com/vrcx-team/VRCX/pull/876

Reply:
let me just re-comment since I linked the wrong thing.
https://github.com/vrcx-team/VRCX/issues/875
API0's content will soon change.

Reply:
We plan to merge nightly and main repo, which will break this.
As stated in the issue above:
- `stable` will contain all releases from the main repository that are not marked as a pre-release.
- `nightly` will contain all releases from the main repository that are marked as a pre-release merged with all releases from the nightly fork.

Reply:
If the source of the API app is the issue, I'll talk with Natsumi about how we want to go forward with that. I'm not opposed to open-sourcing it.

Reply:
Why would that change require a server? The `latest` endpoint according to docs will always return a non pre-release, meanwhile releases itself contains the `prerelease` field.

Of course to do this here it'd require a JSON parser and some logic, which isn't necessarily a problem. But I doubt the API was introduced for the convenience of the Linux script ^^'

Reply:
If you want to mimic the API in the Linux script itself, sure be my guest. This was primarily for parity with the VRCX app updater itself which is already calling api0, along with the legacy domain now pointing towards it.

I'd prefer it to have the vrcx.app server call for the sake of consistency, so if there's anything you'd want that would make you comfortable with it, I'd be more than willing to make amends.

Reply:
It's more or less that this is swapping to a third party server for seemingly no tangible gain on the Linux side.

An obvious convenience advantage here would be if you could pass the release URL directly instead of us needing to parse it here per grep.

On the paranoid side, cheapest thing would obviously be to have the API be auditable by open sourcing the server side.

A way to fix the API trust issue would be to pull the SHA256 from Github and check that, however _SHA256SUMS.txt_ only contains the one for the `VRCX_*_Setup.exe` file :^)

Reply:
Not opposed to open-sourcing, just want to discuss with Natsumi how we want to handle repository ownership then once I clean up the code a little.

The main reason for the switch is the plan for a single repository rather than having a fork for nightlies.
If GitHub had a proper way to filter by release/pre-release, this wouldn't have been an issue.

Do be aware of the following though when you try to mimic the API itself:
- Will need to do pagination, as there wouldn't be a guarantee of the prerelease existing on the first page.
- Also the transition to a single repo won't be immediate, so you'd need to juggle both repos for the latest nightly.
- Github rate limits for unauthenticated requests are about 60/hr, which you could hit often.

Reply:
You could theoretically just get the release ID from the API, query the GitHub API for the specific release, and grab the DL from that if open-sourcing isn't enough.

Reply:
And, to address the convenience factor of the Linux script, I don't mind expanding the API to make it easier for this script to function. One of the goals of this 'update' was parity with the previous API to not break older clients, so as long as that's still maintained, I'm not opposed to new endpoints.

Reply:
The releases endpoint seems to be sorted by the latest releases, and allows returning a maximum of 100 releases. 100 releases without a pre-release seems like a weird scenario.

Github ratelimits are a problem, yeah, but there's not much we can do about that aside fail. I don't think unattendedness was an explicit goal of the script.

> Also the transition to a single repo won't be immediate, so you'd need to juggle both repos for the latest nightly.

I don't have anything nice to say about this.

Anyway yeah, if those two are done, I think the change would be fine. Obviously my authority here is phony but that's just my opinions on it.

Reply:
Source: https://github.com/DubyaDude/VRCX-API
I'll likely have it moved under the `vrcx-team` GitHub org soon.
The only change I had planned from this is to do proper configuration as it's currently hooking into a configuration file used by some of the other apps on the server.

Reply:
It's been moved to https://github.com/vrcx-team/VRCX-API
I've made some of the discussion points into issues, feel free to tack on things if interested: https://github.com/vrcx-team/VRCX-API/issues